src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,170 @@
 package core.basesyntax;
 
+//import jdk.internal.util.ArraysSupport;
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
+
 public class ArrayList<T> implements List<T> {
+
+
+    private static final int def_capacity = 10;
+    private static final Object[] empty_arraydata = {};
```

Comment:
This is wrong.   We need to use field "private int indexOfFirstElement = 0;"
This field can be used to quickly get the index of the first free element in the array, it is generally updated whenever an element is added to or removed from the array.